### ISSUE TITLE
accuracy evaluation

### ISSUE THREAD

**Issue by alanyuchenhou** [yuchenhou/rat#4]
Reevaluate the prediction accuracy:
- for each experiment, divide the data set into a positive subset and a negative subset
- measure the classification accuracy on the positive subset and the negative subset separately

**Comment by ghost** [comment]
Just to clarify, here is what I am wanting (given window sizes W = 200, 500 and 1000):
1. Generate all positive examples P, i.e., non-overlapping subsequences of size W within all DMR regions.
2. Generate all negative examples N, i.e., non-overlapping subsequences of size W within all non-DMR regions.
3. Train a DL classifier using P and N (or some subset of them).
4. Test classifier on all P and N.
5. Report size of P, size of N, number of examples in P incorrectly classified as non-DMR (i.e., false negatives), and number of examples in N incorrectly classified as DMR (i.e., false positives).

**Comment by alanyuchenhou** [comment]
Got it. Thanks for the clarification!

**Comment by alanyuchenhou** [comment]
The confusion matrix for W = 1000:

|  | 0 | 1 |
| --- | --- | --- |
| 0 | 53207 | 0 |
| 1 | 215 | 0 |

Training and testing on all examples, the classifier always gives negative predictions. The accuracy is still high but the result is not very useful. This needs more work (e.g. class weight).

**Comment by ghost** [comment]
Not surprising. I would still like to do one more experiment. Basically what you did before, i.e., same number of positive and negative examples (i.e., undersample the majority class), but apply the learned classifier on all the original examples. I think you got the best result earlier with W=500, so we can just do that window size.

**Comment by alanyuchenhou** [comment]
I adjusted the experiment settings:
- training set: all positive segments and an equal number of negative segments
- test set: all segments
- segment length: 500 bases

And in order to make sure the classifier is actually learning I decided to keep track of training confusion matrix as well. So I got these:
- training

|  | 0 | 1 |
| --- | --- | --- |
| 0 | 89 | 341 |
| 1 | 62 | 368 |
- testing

|  | 0 | 1 |
| --- | --- | --- |
| 0 | 20745 | 85666 |
| 1 | 62 | 368 |

Now the classifier handles positive samples better than negative ones. But the big problem is the training is not as successful as it used to be. So I trained it for much longer and I got this:
- training

|  | 0 | 1 |
| --- | --- | --- |
| 0 | 423 | 7 |
| 1 | 354 | 76 |
- testing

|  | 0 | 1 |
| --- | --- | --- |
| 0 | 99623 | 6788 |
| 1 | 354 | 76 |

Now that the classifier became completely different: it handles negative samples well but not positive ones. But still the training is not very successful. These are very strange results - not sure what's wrong.

**Comment by ghost** [comment]
Okay, thanks. Perhaps the problem is not well suited to this approach. I think we've spent enough time on it for now, but you are welcome to keep playing with it.

**Comment by alanyuchenhou** [comment]
Haven't got much improvements or insights with CNN on the current problem definition. Will consider redefining the problem, e.g.:
- input: the entire sequence of bases
- output: the sequence of labels for each base
